### PR TITLE
Add support for resuming partial upload

### PR DESCRIPTION
**Pull request by sjanc**
This allows uploader to resume partial upload by reusing same SHA hash
in new upload. This patch is based on similar functionality in newtmgr
and is compatible with it.

Signed-off-by: Szymon Janc <szymon.janc@codecoup.pl>

### CODE REVIEW

**Comment by carlescufi** [comment]
cc @nvlsianpu

**Comment by sjanc** [comment]
This is already handled by clearing img_mgmt_ctxt.uploading on erase.

**Comment by nvlsianpu** [comment]
Right now under review from my side.

File: cmd/img_mgmt/src/img_mgmt.c [hunk 1]
```diff
@@ -270,26 +279,35 @@ img_mgmt_encode_upload_rsp(struct mgmt_ctxt *ctxt, int status)
...
     struct image_header hdr;
-    int rc;
 
-    if (len < sizeof hdr) {
+    if (len < sizeof(hdr)) {
         return MGMT_ERR_EINVAL;
     }
 
-    memcpy(&hdr, req_data, sizeof hdr);
+    memcpy(&hdr, req_data, sizeof(hdr));
```
[the remark this answers is not in the document]
Comment (sjanc) [reply cmd/img_mgmt/src/img_mgmt.c]: struct image_header is not packed so we cannot cast to it (and I didn't wanted to change that in this PR)

**Comment by sjanc** [comment]
and client code doesn't require any changes as it already handles responding with offset passed in last packet (which we use here to resume upload)

File: cmd/img_mgmt/src/img_mgmt.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (nvlsianpu) [reply cmd/img_mgmt/src/img_mgmt.c]: sure, understand

**Pull request merged**